Zend/zend_operators.c

```diff
@@ -138,16 +140,35 @@ ZEND_API const unsigned char zend_toupper_map[256] = {
 
 ZEND_API zend_long ZEND_FASTCALL zend_atol(const char *str, size_t str_len) /* {{{ */
 {
+	char *digits_end = NULL;
+
 	if (!str_len) {
 		str_len = strlen(str);
 	}
 
+	/* Ignore trailing whitespace */
+	while (str_len && ZEND_IS_WHITESPACE(str[str_len-1])) --str_len;
+	if (!str_len) return 0;
+
 	/* Perform following multiplications on unsigned to avoid overflow UB.
 	 * For now overflow is silently ignored -- not clear what else can be
 	 * done here, especially as the final result of this function may be
 	 * used in an unsigned context (e.g. "memory_limit=3G", which overflows
```

Comment:
Thank you for your review @TysonAndre 

> The implementation looks good. I've accidentally made the mistake GB in the past, which is ignored, and a warning would be useful

All credit goes to @sgolemon who did the original PR

> I wonder if it'd also make sense to warn about overflow (including after multiplication for suffixes) at the same time, with this change making this helper possibly emit a warning

Agreed, this seems like a good idea. Do you think it should be in this PR or a separate PR ? I would have a preference for a separate PR

Reply:
> The implementation looks good

It also seemed like emitting a warning was safe in all the places that did this, as far as I could tell - e.g. nothing opcache would think was a side effect free opcode would use zend_strtol that I could find and start having side effects/throwing

> All credit goes to sgolemon who did the original PR

I looked at that while reviewing this, just confirming that it still looks good after a rebase.

> Agreed, this seems like a good idea. Do you think it should be in this PR or a separate PR ? I would have a preference for a separate PR

Separate PR if the suffix checking is uncontroversial, same PR if it ended up requiring an RFC.

https://externals.io/message/105646 didn't have much feedback (I don't remember seeing the email, though) - negative or otherwise. Hopefully there are no objections.

Another email should be sent to see if opinions had changed or to describe the latest changed behavior in more details, including

- severity (E_WARNING)
- Error message
- Affected functions (ini_set, ini parsing, ffi, upload progress, etc., and how they changed)
- How this will affect future php versions (I'm assuming none and this will continue emitting an E_WARNING, I can't see how else this would be done (throwing wouldn't work for ini settings))